apps/web/src/screens/Setting/components/MyWallets/Wallet.tsx

```diff
@@ -1,45 +1,82 @@
-import { FC } from 'react';
+import { type FC } from 'react';
+import { Image, StyleSheet } from 'react-native';
+import { Notification } from '@walless/app';
 import { shortenAddress } from '@walless/core';
+import {
+	BindDirections,
+	Hoverable,
+	modalActions,
+	Text,
+	View,
+} from '@walless/gui';
+import { Copy } from '@walless/icons';
 import { PublicKeyDocument } from '@walless/store';
-import { Image, Stack, Text } from '@walless/ui';
 import { getNetworkInfo } from 'utils/helper';
 
 interface Props {
 	item: PublicKeyDocument;
 	index: number;
 }
 
-const Wallet: FC<Props> = ({ item, index }) => {
+export const Wallet: FC<Props> = ({ item, index }) => {
 	const network = getNetworkInfo(item.network);
+	const onCopy = () => {
+		navigator.clipboard.writeText(item._id as string);
+
+		const StyledCopy = () => <Copy size={18} color="#FFFFFF" />;
+
+		modalActions.show({
+			id: 'copied_announcement',
+			component: Notification,
+			bindingDirection: BindDirections.InnerTopRight,
+			positionOffset: { x: 0, y: 20 },
+			maskActiveOpacity: 0,
+			context: {
+				prefix: StyledCopy,
+				message: 'Copied',
+			},
+		});
+
+		setTimeout(() => {
+			modalActions.hide('copied_announcement');
+		}, 1000);
```

Comment:
off-load this method from Presentation layer, should be a shared Valtio action instead, what we do is to call it here instead.